flyteadmin/pkg/manager/impl/util/filters.go

```diff
@@ -318,3 +319,88 @@ func GetNodeExecutionIdentifierFilters(
 	}
 	return append(workflowExecutionIdentifierFilters, nodeIDFilter), nil
 }
+
+// GetIsolationFilter takes in a target resource depth, a user's resource scopes, and the resource's columns and generates
+// appropriate where clauses to filter resources based on a user's isolation context.
+func GetIsolationFilter(ctx context.Context, resourceDepth isolation.TargetResourceScopeDepth, columns common.ResourceColumns) common.IsolationFilter {
+	authzCtx := isolation.IsolationContextFromContext(ctx)
+	resourceScopes := authzCtx.GetResourceScopes()
+
+	// authz is disabled or has been bypassed
+	if len(resourceScopes) == 0 {
+		return nil
+	}
+
+	adjustedResourceScopes := []isolation.ResourceScope{}
+
+	for _, resourceScope := range resourceScopes {
+		if resourceScope.Project == "" {
+			// User has wildcard access
+			return nil
+		}
+
+		tempResourceScope := isolation.ResourceScope{
+			Project: resourceScope.Project,
+		}
+
+		if resourceDepth == isolation.ProjectTargetResourceScopeDepth {
```

Comment:
This is another key implementation detail. If a user has a policy for a scope that includes both the project and domain (ie. flytesnacks:development) and we are making queries against a resource that only has project level scope (ie. projects) we must truncate the user's scope down to the project level (ie. flytesnacks) for this request.